include/copy.h

```diff
@@ -18,10 +18,12 @@ class Copy {
 
     void reader(int buffer_size);
 
+    static std::exception_ptr exceptionPtr;
```

Comment:
with such a design the class is not ready to work with multiple instances of it since this variable is static.
Please re-design the class to allow to use of multiple instances of it.
For this consider std::future/promise https://en.cppreference.com/w/cpp/thread/future